evm/v0.5/testv2/Coordinator.test.ts

```diff
@@ -0,0 +1,909 @@
+          ),
+        )
+        assert.equal(said, sAID)
+      })
+    })
+
+    describe('with an invalid oracle signature', () => {
+      it('saves no service agreement struct, if signatures invalid', async () => {
+        const serviceAgreement = await newServiceAgreement({
+          oracles: [roles.oracleNode.address],
+          aggregator: emptyAggregator.address,
+        })
+        const sAID = h.generateSAID(serviceAgreement)
+        const badOracleSignature = await h.personalSign(sAID, roles.stranger)
+        const badRequestDigestAddr = h.recoverPersonalSignature(
+          sAID,
+          badOracleSignature,
+        )
+        assert.equal(roles.stranger.address, badRequestDigestAddr)
+        const conbinedSignatures = h.combineOracleSignatures([
+          badOracleSignature,
+        ])
+        await h.assertActionThrows(
+          coordinator.initiateServiceAgreement(
+            h.encodeServiceAgreement(serviceAgreement),
+            h.encodeOracleSignatures(conbinedSignatures),
+          ),
+        )
+        await h.checkServiceAgreementAbsent(coordinator, sAID)
+      })
+    })
+
+    describe('Validation of service agreement deadlines', () => {
+      it('Rejects a service agreement with an endAt date in the past', async () => {
+        const serviceAgreement = await newServiceAgreement({
+          endAt: 1,
+          aggregator: emptyAggregator.address,
+        })
+        const sAID = h.generateSAID(serviceAgreement)
+        await h.assertActionThrows(
+          initiateServiceAgreement(coordinator, serviceAgreement),
+        )
+        await h.checkServiceAgreementAbsent(coordinator, sAID)
+      })
+    })
+  })
+
+  describe('#oracleRequest', () => {
+    const to = '0x80e29acb842498fe6591f020bd82766dce619d43'
+    let agreement: h.ServiceAgreement, fHash: string, sAID: string
+
+    beforeEach(async () => {
+      fHash = h.getFunctionSignature(getterSetter, 'requestedBytes32')
+      agreement = await newServiceAgreement({
+        aggregator: meanAggregator.address,
+      })
+      sAID = h.generateSAID(agreement)
+      await initiateServiceAgreement(coordinator, agreement)
+      await link.transfer(roles.consumer.address, h.toWei('1000'))
+    })
+
+    describe('when called through the LINK token with enough payment', () => {
+      it('logs an event', async () => {
+        const payload = h.executeServiceAgreementBytes(sAID, to, fHash, '1')
+
+        link = link.connect(roles.consumer)
+        const tx = await link.transferAndCall(
+          coordinator.address,
+          agreement.payment,
+          payload,
+        )
+        const log = await h.getLog(tx, 2)
+        const addressLog = log.address
+        assert.equal(coordinator.address, addressLog)
+        // If updating this test, be sure to update
+        // services.RunLogTopic20190207withoutIndexes. (Which see for the
+        // calculation of this hash.)
+        const eventSignature =
```

Comment:
We can extract the event signature directly from the contract interface, could we do that instead of hard coding it? It would prevent us having to manually update anything